django/imdb_app/imdb_api/views.py

```diff
@@ -0,0 +1,32 @@
+from django.shortcuts import render
+from .serializers import UserSerializer
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.permissions import AllowAny
+from rest_framework import permissions, viewsets
+from .models import User
+
+# Create your views here.
+
+class UserViewSet(viewsets.ModelViewSet):
+    """
+    API endpoint that allows users to be viewed or edited.
+    """
+    queryset = User.objects.all().order_by('-date_joined')
+    serializer_class = UserSerializer 
+    #permission_classes = [permissions.IsAuthenticated]
+
+class CustomUserRegister(APIView):
+    permission_classes = [AllowAny]
+
+    def post(self, request):
+        reg_serializer = UserSerializer(data=request.data)
+        if reg_serializer.is_valid():
```

Comment:
```suggestion
        if reg_serializer.is_valid(raise_exception=True):
```
i ne moras if -else nego try-except